[TEST] API tests related to the 'test_networks' are failing on Jenkins

## What's the test to develop? Please describe

API tests `harvester_e2e_tests/apis/test_networks.py::TestNetworks::test_create, test_get, test_delete` are failing on Jenkins with below error.
```
    def test_create(self, api_client, unique_name):
        code, data = api_client.networks.create(unique_name, VLAN_ID)
    
>       assert 201 == code, (code, data)
E       AssertionError: (422, {'apiVersion': 'v1', 'code': 422, 'details': {'causes': [{'field': 'metadata.labels', 'message': 'Invalid value:...p': 'k8s.cni.cncf.io', 'kind': 'NetworkAttachmentDefinition', 'name': '11-14-07h19m43s563559'}, 'kind': 'Status', ...})
E       assert 201 == 422

harvester_e2e_tests/apis/test_networks.py:67: AssertionError
```
```
def test_get(self, api_client, unique_name):
        # Case 1: get all vlan networks
        code, data = api_client.networks.get()
    
        assert 200 == code, (code, data)
>       assert len(data['items']) > 0, (code, data)
E       AssertionError: (200, {'apiVersion': 'k8s.cni.cncf.io/v1', 'items': [], 'kind': 'NetworkAttachmentDefinitionList', 'metadata': {'continue': '', 'resourceVersion': '33660'}})
E       assert 0 > 0
E        +  where 0 = len([])

harvester_e2e_tests/apis/test_networks.py:74: AssertionError
```
```
def test_delete(self, api_client, unique_name, wait_timeout):
        code, data = api_client.networks.delete(unique_name)
    
>       assert 200 == code, (f"Failed to delete vlan with error: {code}, {data}")
E       AssertionError: Failed to delete vlan with error: 404, {'kind': 'Status', 'apiVersion': 'v1', 'metadata': {}, 'status': 'Failure', 'message': 'network-attachment-definitions.k8s.cni.cncf.io "11-14-07h19m43s563559" not found', 'reason': 'NotFound', 'details': {'name': '11-14-07h19m43s563559', 'group': 'k8s.cni.cncf.io', 'kind': 'network-attachment-definitions'}, 'code': 404}
E       assert 200 == 404

harvester_e2e_tests/apis/test_networks.py:85: AssertionError
```

--- a/apiclient/harvester_api/managers.py
+++ b/apiclient/harvester_api/managers.py
@@ -3,6 +3,8 @@ from weakref import ref
 from pathlib import Path
 from collections.abc import Mapping
 
+from pkg_resources import parse_version
+
 DEFAULT_NAMESPACE = "default"
 
 
@@ -409,13 +411,19 @@ class NetworkManager(BaseManager):
     API_VERSION = "k8s.cni.cncf.io/v1"
     _KIND = "NetworkAttachmentDefinition"
 
-    def create_data(self, name, namespace, vlan_id):
+    def _default_cluster_network(self):
+        if self.api.cluster_version >= parse_version("v1.1.0"):
+            return "mgmt-br"
+        else:
+            return "harvester-br0"
+
+    def create_data(self, name, namespace, vlan_id, bridge_name):
         data = {
             "apiVersion": self.API_VERSION,
             "kind": self._KIND,
             "metadata": {
                 "labels": {
-                    "networks.harvesterhci.io/type": "L2VlanNetwork"
+                    "network.harvesterhci.io/type": "L2VlanNetwork"
                 },
                 "name": name,
                 "namespace": namespace
@@ -425,7 +433,7 @@ class NetworkManager(BaseManager):
                     "cniVersion": "0.3.1",
                     "name": name,
                     "type": "bridge",
-                    "bridge": "harvester-br0",
+                    "bridge": bridge_name,
                     "promiscMode": True,
                     "vlan": vlan_id,
                     "ipam": {}
@@ -438,8 +446,10 @@ class NetworkManager(BaseManager):
         path = self.PATH_fmt.format(uid=name, ns=namespace, NETWORK_API=self.API_VERSION)
         return self._get(path, raw=raw)
 
-    def create(self, name, vlan_id, namespace=DEFAULT_NAMESPACE, *, raw=False):
-        data = self.create_data(name, namespace, vlan_id)
+    def create(self, name, vlan_id, namespace=DEFAULT_NAMESPACE, *,
+               cluster_network=None, raw=False):
+        cluster_network = cluster_network or self._default_cluster_network()
+        data = self.create_data(name, namespace, vlan_id, cluster_network)
         path = self.PATH_fmt.format(uid="", ns=namespace, NETWORK_API=self.API_VERSION)
         return self._create(path, json=data, raw=raw)
 

--- a/harvester_e2e_tests/apis/test_networks.py
+++ b/harvester_e2e_tests/apis/test_networks.py
@@ -45,6 +45,7 @@ class TestNetworksNegative:
         assert "NotFound" == data.get("reason"), (code, data)
 
     @pytest.mark.parametrize("vlan_id", [0, 4095])
+    @pytest.mark.skip_version_after("v1.0.3")  # ref to harvester/issues/3151
     def test_create_with_invalid_id(self, api_client, unique_name, vlan_id):
         code, data = api_client.networks.create(unique_name, vlan_id)
 
@@ -61,11 +62,13 @@ class TestNetworksNegative:
 @pytest.mark.p0
 @pytest.mark.networks
 class TestNetworks:
+
+    @pytest.mark.dependency(name="create_network")
     def test_create(self, api_client, unique_name):
         code, data = api_client.networks.create(unique_name, VLAN_ID)
-
         assert 201 == code, (code, data)
 
+    @pytest.mark.dependency(depends=["create_network"])
     def test_get(self, api_client, unique_name):
         # Case 1: get all vlan networks
         code, data = api_client.networks.get()
@@ -79,6 +82,7 @@ class TestNetworks:
         assert 200 == code, (code, data)
         assert unique_name == data['metadata']['name'], (code, data)
 
+    @pytest.mark.dependency(depends=["create_network"])
     def test_delete(self, api_client, unique_name, wait_timeout):
         code, data = api_client.networks.delete(unique_name)
 

